### DISCUSSION TITLE
Being able to open an image from memory (og.Images.open)

### DISCUSSION THREAD

**Discussion by MadMenHitBooker** [microsoft/onnxruntime-genai#845] [New features / APIs]
**Describe the problem**
It would be interesting to being able to load and image for the phi3vision from memory, or as a cv2 or PIL image.

Currently as showed in the example the loading is being done like this.

```python
image = og.Images.open(image_path)
```

**Comment by baijumeswani** [comment]
We did receive a request for this earlier as well. I will add a way to load an image from memory.

**Reply by JehanJaye** [reply]
Is this feature already added or is it currently work in progress?

**Comment by kunal-vaishnavi** [comment]
Support for loading images from memory has been added [here](https://github.com/microsoft/onnxruntime-genai/pull/796). Here is an example.

```python
with open(image_path, "rb") as image:
    bytes = image.read()
    images = og.Images.open_bytes(bytes)
```
